src/lib/operators/aggregate_sort.cpp

```diff
@@ -0,0 +1,416 @@
+#include "aggregate_sort.hpp"
+
+#include <memory>
+#include <vector>
+
+#include "aggregate/aggregate_traits.hpp"
+#include "all_type_variant.hpp"
+#include "constant_mappings.hpp"
+#include "operators/sort.hpp"
+#include "storage/segment_iterate.hpp"
+#include "table_wrapper.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+AggregateSort::AggregateSort(const std::shared_ptr<AbstractOperator>& in,
+                             const std::vector<AggregateColumnDefinition>& aggregates,
+                             const std::vector<ColumnID>& groupby_column_ids)
+    : AbstractAggregateOperator(in, aggregates, groupby_column_ids) {}
+
+const std::string AggregateSort::name() const { return "AggregateSort"; }
+
+template <typename ColumnType, typename AggregateType, AggregateFunction function>
+void AggregateSort::_aggregate_values(std::set<RowID>& aggregate_group_offsets, uint64_t aggregate_index,
+                                      AggregateFunctor<ColumnType, AggregateType> aggregate_function,
+                                      std::shared_ptr<const Table> sorted_table) {
+  const size_t num_groups = aggregate_group_offsets.size() + 1;
+  auto aggregate_results = std::vector<AggregateType>(num_groups);
+  auto aggregate_null_values = std::vector<bool>(num_groups, false);
+
+  uint64_t value_count = 0u;
```

Comment:
Some of the variables are indeed aggregate-specific. However, making several special cases (and const expressions) for the several aggregate functions would lead to worse readability. We believe markus comment from abstract_aggregate_operator.hpp:93 applies here too.